cnd/src/swap_protocols/ledger_states.rs

```diff
@@ -0,0 +1,126 @@
+use crate::swap_protocols::{
+    rfc003::{create_swap::SwapEvent, LedgerState},
+    state::{Get, Insert, Update},
+    swap_id::SwapId,
+};
+use std::{
+    any::Any,
+    collections::HashMap,
+    ops::{Deref, DerefMut},
+    sync::Mutex,
+};
+
+#[derive(Default, Debug)]
+pub struct AlphaLedgerState {
+    states: Mutex<HashMap<SwapId, Box<dyn Any + Send>>>,
+}
+
+#[derive(Default, Debug)]
+pub struct BetaLedgerState {
```

Comment:
:mega: The two structs are exactly the same ?! We can we not just have two instances of the same struct?

Reply:
I thought it was on purpose, as it is part of this ticket to achieve the split between alpha and beta. Given that there might be specific parameters for each ledger we might need different structs. I see how it could be one here, but would it make sense to keep it for later changes on top of this?
Additionally I wonder how this will go together with introducing new protocols.

Reply:
I believe it was a mistake on my part. If we continue using this model, there won't be any difference between how we store alpha and beta ledger (protocol) state in memory: it will always be `Any`. It's the implementation of `Update` that will vary depending on the protocol.